src/main/java/hu/wtomee/objects/DFA.java

```diff
@@ -0,0 +1,158 @@
+package hu.wtomee.objects;
+
+import java.security.InvalidParameterException;
+import java.util.List;
+
+public class DFA {
+
+    List<String> states;
+
+    List<String> alphabet;
+
+    List<Transition> transitions;
+
+    String initialState;
+
+    List<String> finalStates;
+
+    String currentState;
+
+    public DFA() {
+    }
+
+    public DFA(List<String> states, List<String> alphabet, List<Transition> transitions, String initialState, List<String> finalStates) {
+        this.states = states;
+        this.alphabet = alphabet;
+        this.transitions = transitions;
+        this.initialState = initialState;
+        this.finalStates = finalStates;
+    }
+
+    public List<String> getStates() {
+        return states;
+    }
+
+    public void setStates(List<String> states) {
+        this.states = states;
+    }
+
+    public List<String> getAlphabet() {
+        return alphabet;
+    }
+
+    public void setAlphabet(List<String> alphabet) {
+        this.alphabet = alphabet;
+    }
+
+    public List<Transition> getTransitions() {
+        return transitions;
+    }
+
+    public void setTransitions(List<Transition> transitions) {
+        this.transitions = transitions;
+    }
+
+    public String getInitialState() {
+        return initialState;
+    }
+
+    public void setInitialState(String initialState) {
+        this.initialState = initialState;
+    }
+
+    public List<String> getFinalStates() {
+        return finalStates;
+    }
+
+    public void setFinalStates(List<String> finalStates) {
+        this.finalStates = finalStates;
+    }
+
+    public String getCurrentState() {
+        return currentState;
+    }
+
+    public void setCurrentState(String currentState) {
+        this.currentState = currentState;
+    }
+    public void checkTransitions() {
+        for(Transition t : transitions) {
+            if (!states.contains(t.nextState) || ! states.contains(t.fromState)) {
+                throw new InvalidParameterException("State doesnt exists in transition: " +  t.toString());
+            }
+            if (!alphabet.contains(t.symbol)){
+                throw new InvalidParameterException("Symbol doesnt exists in transition: " + t.toString());
+            }
+
+        }
+
+        System.out.println("OK! Transitions checked! ");
+    }
+
+    public DFA resetState() {
+        currentState = initialState;
+        return this;
+    }
+
+    public DFA input(String input) {
```

Comment:
kisse confusing, hogy a fuggveny neve es az argumentum neve ugyanaz. 
a fuggvenyt nem is ertem a neve alapjan, hogy mit csinalhat, az argumentum meg csak egy symbol szerintem, ha nem `List<String>` nem pedig a teljes input.

Reply:
A célja az lett volna tényleg, hogy egy karaktert vizsgáljon és egyébként char-t is adtam volna át, de szépen Stringesítettem.
Resolváltam :)